### ISSUE TITLE
Record the number of cores in the CPU set

### ISSUE THREAD

**Issue by bbockelm** [cms-sw/cmssw#15652]
We recently hit a case (NERSC) where the pilot job was misconfigured and all CMSSW instances were set to run on a single core.

In the FJR, in the `SystemCPU` section, we should report the number of CPUs in the current CPU set.  That would have allowed us to notice the situation.

If we really want to go wild, we could record a warning any time the number of CPUs is less than the number of configured TBB threads.

Looking up the number of CPUs is fairly straightforward.  Here's an example program:

```
#define _GNU_SOURCE
#include <sched.h>
#include <errno.h>
#include <stdio.h>

int main(int argc, char *argv[]) {

  cpu_set_t *cpusetp;

  unsigned current_size = 128;
  unsigned cpu_count = 0;
  while (current_size*2 > current_size) {
    cpusetp = CPU_ALLOC(current_size);
    CPU_ZERO_S(CPU_ALLOC_SIZE(current_size), cpusetp);

    if (sched_getaffinity(0, CPU_ALLOC_SIZE(current_size), cpusetp)) {
      if (errno == EINVAL) {
        current_size *= 2;
        CPU_FREE(cpusetp);
        continue;
      }
      perror("schedd_getaffinity");
    }
    cpu_count = CPU_COUNT_S(CPU_ALLOC_SIZE(current_size), cpusetp);
    CPU_FREE(cpusetp);
    break;
  }

  printf("CPU set count: %d\n", cpu_count);

  return 0;
}
```

**Comment by bbockelm** [comment]
For example, we currently have:
```
<PerformanceReport>
  <PerformanceSummary Metric="SystemCPU">
    <Metric Name="CPUModels" Value="Intel(R) Xeon(R) CPU E5-2695 v2 @ 2.40GHz"/>
    <Metric Name="averageCoreSpeed" Value="2401"/>
    <Metric Name="totalCPUs" Value="48"/>
  </PerformanceSummary>
</PerformanceReport>
```

I would like to see:
```
<PerformanceReport>
  <PerformanceSummary Metric="SystemCPU">
    <Metric Name="CPUModels" Value="Intel(R) Xeon(R) CPU E5-2695 v2 @ 2.40GHz"/>
    <Metric Name="averageCoreSpeed" Value="2401"/>
    <Metric Name="totalCPUs" Value="48"/>
    <Metric Name="cpusetCount" Value="4"/>
  </PerformanceSummary>
</PerformanceReport>
```

**Comment by Dr15Jones** [comment]
assign core

**Comment by bbockelm** [comment]
Fixed by PR #15760.

**Comment by smuzaffar** [comment]
+core
fixed by  #15760
